internal/cli/certificate.go

```diff
@@ -0,0 +1,434 @@
+	"github.com/flightctl/flightctl/internal/client"
+	fccrypto "github.com/flightctl/flightctl/internal/crypto"
+	"github.com/flightctl/flightctl/internal/util/validation"
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/google/uuid"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"golang.org/x/term"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/yaml"
+)
+
+const secondsInDay = 86400
+const agentPath = "certs/"
+
+type CertificateOptions struct {
+	GlobalOptions
+	Name         string
+	Expiration   string
+	OutputFormat string
+	OutputDir    string
+	EncryptKey   bool
+	SignerName   string
+}
+
+func DefaultCertificateOptions() *CertificateOptions {
+	return &CertificateOptions{
+		GlobalOptions: DefaultGlobalOptions(),
+		Expiration:    "7d",
+		OutputFormat:  "reference",
+		OutputDir:     "./",
+		EncryptKey:    false,
+		SignerName:    "enrollment",
+	}
+}
+
+func NewCmdCertificate() *cobra.Command {
+	o := DefaultCertificateOptions()
+	cmd := &cobra.Command{
+		Use: "certificate request [flags]",
+		// more subcommands may be added later
+		Short:     "request a new enrollment certificate for a device with 'certificate request'",
+		Args:      cobra.MatchAll(cobra.MinimumNArgs(1), cobra.OnlyValidArgs),
+		ValidArgs: []string{"request"},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			if err := o.Validate(args); err != nil {
+				return err
+			}
+			return o.Run(cmd.Context(), args)
+		},
+		SilenceUsage: true,
+	}
+	o.Bind(cmd.Flags())
+	return cmd
+}
+
+func (o *CertificateOptions) Bind(fs *pflag.FlagSet) {
+	o.GlobalOptions.Bind(fs)
+	fs.StringVarP(&o.Name, "name", "n", "csr", "specify a name for the certificate signing request")
+	fs.StringVarP(&o.Expiration, "expiration", "x", "7d", "specify desired certificate expiration in days, example: 7d")
+	fs.StringVarP(&o.OutputFormat, "output-format", "o", "default", "specify desired output format for an enrollment cert: either 'reference' to have the config file reference key and cert file paths, or 'embedded' to have the key and cert embedded in the config file")
+	fs.StringVarP(&o.OutputDir, "output-dir", "d", "./", "specify desired output directory for key, cert, and ca files (defaults to current directory)")
+	fs.StringVarP(&o.SignerName, "cert-type", "t", "enrollment", "specify the type of certificate requested: 'enrollment' or 'ca' (default 'enrollment')")
+	fs.BoolVarP(&o.EncryptKey, "encrypt", "s", false, "option to encrypt key file with a password from env var $FCPASS, or if $FCPASS is not set password must be provided during runtime")
+}
+
+func (o *CertificateOptions) Complete(cmd *cobra.Command, args []string) error {
+	return o.GlobalOptions.Complete(cmd, args)
+}
+
+func (o *CertificateOptions) Validate(args []string) error {
+	if err := o.GlobalOptions.Validate(args); err != nil {
+		return err
+	}
+
+	errs := validation.ValidateSignerName(o.SignerName)
```

Comment:
I will probably pull in the server-side validation code, including all deps from K8s. We might want to replace that in a later PR.